src/nvim/runtime.c

```diff
@@ -49,7 +50,11 @@ void ex_runtime(exarg_T *eap)
 
 static void source_callback(char_u *fname, void *cookie)
 {
-  (void)do_source(fname, false, DOSO_NONE);
+  if (path_with_extension((const char *)fname, "lua")) {
+    nlua_exec_file((const char *)fname);
+  } else {
+    (void)do_source(fname, false, DOSO_NONE);
+  }
```

Comment:
what if you change `do_source` to be the one that handles checking the path?

I see this exact if statement more than once in the code, which is worrisome that we'll miss something.

Reply:
Those statements are used in 2 places . I didn't want to change `do_source` beacause it can affect a lot of other stuff that I'm probably not aware of. For example `:source` of lua files will actually work .

Reply:
hmm my gut feeling is that _if_ `runtime some/file.lua` works then so should _also_ `source also/some/file.lua`

Reply:
Ok . I'll move the funtionality to `do_source` Since you both want it. Well it's an extension check . Recognizing lua files as lua files instead of vim can't break anything I think :)

Reply:
Ok done . Now you can do source on lua files too . I've looked at all the uses of do_source . It seems to be a harmless addition :)

Reply:
> Ok . I'll move the funtionality to `do_source` Since you both want it. Well it's an extension check . Recognizing lua files as lua files instead of vim can't break anything I think :)

What if there's no extension on the file?

Reply:
Assumed vil like now